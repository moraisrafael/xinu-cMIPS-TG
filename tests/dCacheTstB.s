	.include "cMIPS.s"
	.text
	.align 2
	.globl _start
	.ent _start
_start: la   $15, x_DATA_BASE_ADDR
	addi $3,$0,-10
	ori  $4,$0,1
	nop
stores:	sb   $3, 0($15)
	addi $3,$3,1
	add  $15,$15,$4
        bne  $3,$0,stores
	nop  # flush pipe
	nop
	nop
	nop
	nop
	la   $15, x_DATA_BASE_ADDR
	la   $26, x_IO_BASE_ADDR
	addi $3,$0,-10
	ori  $4,$0,1
loads:	lb   $8, 0($15)
	add  $15,$15,$4
	sb   $8, 0($26)
	addi $3,$3,1
	add  $26,$26,$4
        bne  $3,$0,loads
        nop
	nop
	nop
	nop
	nop
        wait
        nop
	nop
	.end _start
