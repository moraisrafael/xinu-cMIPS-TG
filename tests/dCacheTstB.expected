fffffff6
fffffff7
fffffff8
fffffff9
fffffffa
fffffffb
fffffffc
fffffffd
fffffffe
ffffffff
